runconfig/parse_unix.go

```diff
@@ -7,51 +7,53 @@ import (
 	"strings"
 )
 
-func parseNetMode(netMode string) (NetworkMode, error) {
-	parts := strings.Split(netMode, ":")
+// ValidateNetMode ensures that the various combinations of requested
+// network settings are valid.
+func ValidateNetMode(c *Config, hc *HostConfig) error {
+	// We may not be passed a host config, such as in the case of docker commit
+	if hc == nil {
+		return nil
+	}
```

Comment:
I was wondering on how other Integration-cli tests were passing without forcing the netmode and this gives the answer :-) . I think it will be cleaner to set the network mode as "default" in the server code if we encounter an empty netmode. instead of returning here.